File: app/src/main/java/com/example/gemini_ai_chat/MessageDao.java

import androidx.room.Dao;
import androidx.room.Insert;
import androidx.room.Query;
import androidx.room.Update;

import java.util.List;

@Dao
public interface MessageDao {
    @Query("SELECT * FROM messages ORDER BY timestamp ASC")
    List<Message> getAllMessages();

    @Insert
    long insert(Message message);

    @Update
    void update(Message message);
}

File: app/src/main/java/com/example/gemini_ai_chat/MessageDao.kt

package com.example.gemini_ai_chat

import androidx.lifecycle.LiveData
import androidx.room.Dao
import androidx.room.Insert
import androidx.room.Query

@Dao
interface MessageDao {
    @Query("SELECT * FROM messages ORDER BY timestamp ASC")
    fun getAllMessages(): LiveData<List<Message>>

    @Insert
    suspend fun insert(message: Message)
}
